crates/biome_js_analyze/src/semantic_analyzers/nursery/use_sorted_classes/any_class_string_like.rs

```diff
@@ -0,0 +1,225 @@
+use biome_analyze::{
+    AddVisitor, AnalyzerOptions, Phases, QueryMatch, Queryable, RuleKey, ServiceBag, Visitor,
+    VisitorContext,
+};
+use biome_js_syntax::{
+    JsCallArguments, JsCallExpression, JsLanguage, JsStringLiteralExpression,
+    JsTemplateChunkElement, JsxAttribute, JsxString,
+};
+use biome_rowan::{
+    declare_node_union, AstNode, Language, SyntaxNode, TextRange, TokenText, WalkEvent,
+};
+
+use super::UseSortedClassesOptions;
+
+// utils
+// -----
+
+fn get_options_from_analyzer(analyzer_options: &AnalyzerOptions) -> UseSortedClassesOptions {
+    match analyzer_options
+        .configuration
+        .rules
+        .get_rule_options::<UseSortedClassesOptions>(&RuleKey::new("nursery", "useSortedClasses"))
+    {
+        Some(options) => options.clone(),
+        None => UseSortedClassesOptions::default(),
+    }
+}
+
+fn get_callee_name(call_expression: &JsCallExpression) -> Option<TokenText> {
+    call_expression
+        .callee()
+        .ok()?
+        .as_js_identifier_expression()?
+        .name()
+        .ok()?
+        .name()
+        .ok()
+}
+
+fn is_call_expression_of_target_function(
+    call_expression: &JsCallExpression,
+    target_functions: &[String],
+) -> bool {
+    match get_callee_name(call_expression) {
+        Some(name) => target_functions.contains(&name.to_string()),
+        None => false,
+    }
+}
+
+fn get_attribute_name(attribute: &JsxAttribute) -> Option<String> {
+    Some(attribute.name().ok()?.as_jsx_name()?.to_string())
+}
```

Comment:
We could avoid an allocation by using `TokenText` instead of `String`.
Moreover, I think you want the trimmed version of the attribute?
```suggestion
fn get_attribute_name(attribute: &JsxAttribute) -> Option<TokenText> {
    Some(attribute.name().ok()?.as_jsx_name()?.value().ok()?.token_text_trimmed())
}
```

Reply:
Good call, wasn't familiar with the internals of TokenText but avoiding string allocations is a nice win. About trimming, idk, not sure that an attribute name can have whitespace around in the first place? But that depends on specifics of Biome parsing and AST that I'm not familiar enough with.

Reply:
Actually it can if [it is on its own line](https://biomejs.dev/playground/?enabledLinting=false&code=PABhAAoAIAAgAGgAcgBlAGYAPQAiACIAPgBsAGkAbgBrAAoAPAAvAGEAPgA%3D).

Reply:
done, but `value` has to be `value_token`